runtime-transaction/src/transaction_meta.rs

```diff
@@ -39,4 +40,5 @@ pub struct TransactionMeta {
     pub(crate) is_simple_vote_transaction: bool,
     pub(crate) signature_details: TransactionSignatureDetails,
     pub(crate) compute_budget_instruction_details: ComputeBudgetInstructionDetails,
+    pub(crate) instruction_data_len: u64,
```

Comment:
could use a u16 here? it cannot be large than packet size. was using a u64 previously so kept same type for this PR.

Reply:
yea, may as well change to `u16`, seems more proper

Reply:
can also save some space by using `u16` in `TransactionSignatureDetails`

Reply:
did the u16 for data len...`TransactionSignatureDetails` is an `sdk` type (imo we should deprecate it anyway), so not changing that in this PR.